lib/ansible/module_utils/vmware.py

```diff
@@ -354,13 +354,11 @@ def gather_vm_facts(content, vm):
         for device in vmnet:
             net_dict[device.macAddress] = list(device.ipAddress)
 
-    for dummy, v in iteritems(net_dict):
-        for ipaddress in v:
-            if ipaddress:
-                if '::' in ipaddress:
-                    facts['ipv6'] = ipaddress
-                else:
-                    facts['ipv4'] = ipaddress
+    if vm.guest.ipAddress:
+        if '::' in vm.guest.ipAddress:
```

Comment:
This is a bit of a bad way of checking wether it is IPv6 or IPv4: `::` only occurs if one of the hexadectets is `0000`. You can easily end up with `2001:db8:1:2:3:4:5:6`, which does not contain `::`. 

quick fix: check for `:` in `ipaddress`. 
Clean fix: use the `ipaddress` python module (python 3.3+ native https://docs.python.org/3/library/ipaddress.html; or https://pypi.org/project/ipaddress/ for 2.6, 2.7 and 3.2, which is a backport (so same functionality).

This is the interesting part:

> ipaddress.ip_address(address)
> Return an IPv4Address or IPv6Address object depending on the IP address passed as argument. Either IPv4 or IPv6 addresses may be supplied; integers less than 2**32 will be considered to be IPv4 by default. A **ValueError** is raised if address does not represent a valid IPv4 or IPv6 address.

```python
>>>ipaddress.ip_address('192.168.0.1')
IPv4Address('192.168.0.1')
>>> ipaddress.ip_address('2001:db8::')
IPv6Address('2001:db8::')
```

Reply:
I'll throw the quick fix in for now, I'll leave deciding on adding a python2 dep to @Akasurde .